src/frameworks/next/utils.ts

```diff
@@ -0,0 +1,103 @@
+import type { Header, Redirect, Rewrite } from "next/dist/lib/load-custom-routes";
+import type { Manifest, RoutesManifestRewrite } from "./interfaces";
+import { isUrl } from "../utils";
+
+/**
+ * Whether the given path has a regex or not.
+ * According to the Next.js documentation:
+ * ```md
+ *  To match a regex path you can wrap the regex in parentheses
+ *  after a parameter, for example /post/:slug(\\d{1,}) will match /post/123
+ *  but not /post/abc.
+ * ```
+ * See: https://nextjs.org/docs/api-reference/next.config.js/redirects#regex-path-matching
+ */
+export function pathHasRegex(path: string): boolean {
+  // finds parentheses that are not preceded by double backslashes
+  return /(?<!\\)\(/.test(path);
+}
+
+/**
+ * Remove double backslashes from a string
+ */
+export function cleanEscapedChars(path: string): string {
+  return path.replace(/\\/g, "");
```

Comment:
@leoortizz `path.replace(/\\([\(\){}\:\+\?])/g, (a,b) => b)` should do it IMO. WDYT?

```
> '/\\(\\)\\{\\}\\:\\+\\?/:slug'.replace(/\\([\(\){}\:\+\?])/g, (a,b) => b)
'/(){}:+?/:slug'
```

Reply:
"The following characters (, ), {, }, :, *, +, ? are used for regex path matching, so when used in the source as non-special values they must be escaped by adding \\ before them"